simulator/infra/macro.h

```diff
@@ -177,6 +177,13 @@ static constexpr T ones_ls(const T& value, size_t shamt)
     return ~(~value << shamt);
 }
 
+template<typename T>
+static constexpr T rotate_l(const T& value, size_t shamt)
+{
+    size_t xlen = sizeof(T) * 8;
+    return (value << shamt) | (value >> ((xlen - shamt) & (xlen - 1)));
+}
+
```

Comment:
I believe we'll merge @ChemrovKirill's request soon where he has this function already w/ unit tests. I suggest to keep his version though.

Reply:
So int this case i should wait for his merging?

Reply:
You can pull from his repository (git pull https://github.com/ChemrovKirill/mipt-mips master)